File: 19-meca_flu-eloudrhiri-jacquet-navez/force.m
function [fx,fy]=force(p,x,y) 
%% CONSTANT(s)
h=0.01;
%% COMPUTATION
fy=0;
fx=0;

for i = 1:length(x)-1
    if x(i) < x(i+1)
        fx = fx + (h)*(p(i)+p(i+1))/2;
    elseif x(i+1) < x(i)
        fx = fx + (-h)*(p(i)+p(i+1))/2;
    end
    
    if y(i) < y(i+1)
        fy = fy + (-h)*(p(i)+p(i+1))/2;
    elseif y(i+1) < y(i)
        fy = fy + (h)*(p(i)+p(i+1))/2;
    end
end 

end 

File: force.py
def force(p, x, y):
    fx = 0
    fy = 0
    h = 0.01

    for i in range(0, len(x) - 1):
        if x[i] < x[i + 1]:
            fx = fx + h * (p[i] + p[i + 1]) / 2

        elif x[i + 1] < x[i]:
            fx = fx + (-h) * (p[i] + p[i + 1]) / 2

        if y[i] < y[i + 1]:
            fy = fy + (-h) * (p[i] + p[i + 1]) / 2

        elif y[i + 1] < y[i]:
            fy = fy + h * (p[i] + p[i + 1]) / 2

    return fx, fy
